File: packages/create-uniform-app/src/context.ts
import { isAbsolute, resolve } from 'path';

import { Context } from './types';

export const resolvePath = (absoluteOrRelativePath: string, relativeTo = process.cwd()) => {
    if (isAbsolute(absoluteOrRelativePath)) return absoluteOrRelativePath;

    return resolve(relativeTo, absoluteOrRelativePath).replace(/\\/g, '/');
};

function getUserPkgManager() {
    // This environment variable is set by npm and yarn but pnpm seems less consistent
    const userAgent = process.env.npm_config_user_agent;

    if (userAgent) {
        if (userAgent.startsWith('yarn')) {
            return 'yarn';
        } else if (userAgent.startsWith('pnpm')) {
            return 'pnpm';
        } else {
            return 'npm';
        }
    } else {
        // If no user agent is set, assume npm
        return 'npm';
    }
}

export const loadContext = async (): Promise<Context> => {
    const yargs = process.argv.slice(2);

    return {
        yargs: yargs,
        packageManager: getUserPkgManager(),
    };
};

File: packages/uniform-cli/src/context.rs
use std::path::{Path, PathBuf};

pub fn resolve_path(absolute_or_relative_path: &str, relative_to: &str) -> String {
    if Path::new(absolute_or_relative_path).is_absolute() {
        return absolute_or_relative_path.to_string();
    }

    let resolved_path = PathBuf::from(relative_to).join(absolute_or_relative_path);
    resolved_path.to_str().unwrap().replace("\\", "/")
}
